src/compile/time.ts

```diff
@@ -49,6 +49,38 @@ export function format(timeUnit, abbreviated = false): string {
   return out.length > 0 ? out.join(' ') : undefined;
 }
 
+/** returns the smallest nice unit for scale.nice */
+export function smallestUnit(timeUnit): string {
+  if (!timeUnit) {
+    return undefined;
+  }
+
+  if (timeUnit.indexOf('second') > -1) {
+    return 'second';
+  }
+
+  if (timeUnit.indexOf('minute') > -1) {
+    return 'minute';
+  }
+
+  if (timeUnit.indexOf('hour') > -1) {
+    return 'hour';
+  }
+
+  if (timeUnit.indexOf('day') > -1 || timeUnit.indexOf('date') > -1) {
```

Comment:
My main concern is that there are many days in the month so day for nice may be too detailed. What exactly happens when we set nice to day anyway? Maybe I'm seeing a problem where no problem is.